### PR TITLE
Adds More Botany Chemical Reactions

### PR DESCRIPTION
**Pull request by bathosbathtime**
This PR adds a series of chemical reactions using botany chemicals.  I saw someone in the thread mention that they miss being able to do chemical reactions inside of plants, so I hope that this might help.  I went down the list of plant chemicals on the wiki so if there is one that isn't listed there it got skipped over.  I also skipped over creatine and fix-o-vien as I wasn't sure how they'd interact inside of a plant.  They can be added if needed.
Also nitroglycerin removes phenol and formic acid since they're used in the reaction now.
EDIT: the weedeater reaction has been removed due to conflicts.

:cl:
 * rscadd: Mirrors over chemical reactions to botany chemicals

### CODE REVIEW

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
I rewrote reagent code so we didn't have to have botany subtypes of normal chem reactions.

Refer to #20652

**Comment by bathosbathtime** [comment]
Ah, I will update these tomorrow to add the new IDs to all of the reactions instead of having new ones.  I even noticed the new IDs in some of the recipes but I was unaware of the purpose.  Thank you.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -1064,6 +1066,223 @@
 			var/mob/mob_holder = holder.my_atom.loc
 			mob_holder.drop_item(holder.my_atom) //Bandaid to work around items becoming invisible when their appearance is changed!
 
+///////////////////////////////////////////////Pharmacognosy Time/////////////////////////////////////////////////
+
+/datum/chemical_reaction/tricordrazine/botany
+	name = "Tricordrazine Botany"
+	id = TRICORDRAZINE
+	result = TRICORDRAZINE
+	required_reagents = list(INAPROVALINE = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1075]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -1064,6 +1066,223 @@
...
+	id = STERILIZINE
+	result = STERILIZINE
+	required_reagents = list(ETHANOL = 1, ALLICIN = 1, CHLORINE = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/hyronalin/botany
+	name = "Hyronalin Botany"
+	id = HYRONALIN
+	result = HYRONALIN
+	required_reagents = list(RADIUM = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1096]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -1064,6 +1066,223 @@
...
+	result_amount = 2
+
+/datum/chemical_reaction/alkysine/botany
+	name = "Alkysine Botany"
+	id = ALKYSINE
+	result = ALKYSINE
+	required_reagents = list(CHLORINE = 1, NITROGEN = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1103]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 4]
```diff
@@ -1064,6 +1066,223 @@
...
+	result_amount = 2
+
+/datum/chemical_reaction/alkycosine/botany
+	name = "Alkycosine Botany"
+	id = ALKYCOSINE
+	result = ALKYCOSINE
+	required_reagents = list(ALKYSINE = 1, BLEACH = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1110]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 5]
```diff
@@ -1064,6 +1066,223 @@
...
+	required_temp = T0C + 40
+	result_amount = 4
+
+/datum/chemical_reaction/imidazoline/botany
+	name = "Imidazoline Botany"
+	id = IMIDAZOLINE
+	result = IMIDAZOLINE
+	required_reagents = list(CARBON = 1, HYDROGEN = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1118]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 6]
```diff
@@ -1064,6 +1066,223 @@
...
+	result_amount = 2
+
+/datum/chemical_reaction/inacusiate/botany
+	name = "Inacusiate Botany"
+	id = INACUSIATE
+	result = INACUSIATE
+	required_reagents = list(WATER = 1, CARBON = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1125]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 7]
```diff
@@ -1064,6 +1066,223 @@
...
+	result_amount = 3
+
+/datum/chemical_reaction/ethylredoxrazine/botany
+	name = "Ethylredoxrazine Botany"
+	id = ETHYLREDOXRAZINE
+	result = ETHYLREDOXRAZINE
+	required_reagents = list(OXYGEN = 1, ALLICIN = 1, CARBON = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1132]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 8]
```diff
@@ -1064,6 +1066,223 @@
...
+	result_amount = 3
+
+/datum/chemical_reaction/mindbreaker/botany
+	name = "Mindbreaker Toxin Botany"
+	id = MINDBREAKER
+	result = MINDBREAKER
+	required_reagents = list(SILICON = 1, HYDROGEN = 1, ALLICIN = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1139]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 9]
```diff
@@ -1064,6 +1066,223 @@
...
+	id = PEPTOBISMOL
+	result = PEPTOBISMOL
+	required_reagents = list(ALLICIN = 1, DISCOUNT = 1)
+	result_amount = 2
+
+/datum/chemical_reaction/sterilizine/botany
+	name = "Sterilizine Botany"
+	id = STERILIZINE
+	result = STERILIZINE
+	required_reagents = list(ETHANOL = 1, ALLICIN = 1, CHLORINE = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1089]: This already works

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 10]
```diff
@@ -1064,6 +1066,223 @@
...
+	id = PACID
+	result = PACID
+	required_reagents = list(FORMIC_ACID = 1, CHLORINE = 1, POTASSIUM = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/glycerol/botany
+	name = "Glycerol Botany"
+	id = GLYCEROL
+	result = GLYCEROL
+	required_reagents = list(CORNOIL = 3, FORMIC_ACID = 1)
```
Comment (MadmanMartian) [inline code/modules/reagents/Chemistry-Recipes.dm:1153]: This already works

**Comment by MadmanMartian** [comment]
A lot of these already work.

**Comment by Kurfursten** [comment]
#21353

**Pull request closed without merging**